src/promise-katas-2.js

```diff
@@ -47,7 +78,20 @@ const dog = () => {};
 // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
 //
 
-const joke = () => {};
+const joke = async () => {
+  try {
+  const { joke } = await fetch("jokes", "question");
+  const { answer } = await fetch("jokes");
+  console.log(joke, answer);
+  return {
+    "question": joke,
+    "answer": answer
+  }
+}
+    catch(data) {
+    console.log(data);
+    }
+  };
```

Comment:
Theres a few things to look at here. First is that while this works, it isnt using the methods the exercise was designed for, namely `promise.all`. The idea here was to build on the fetch/then pattern from above, and build on it with a new feature. It would be a good idea to revisit this again, as its an important topic and a very helpful one to understand. Here is what this function could look like using the promise all appraoch:

```
  return Promise.all([
    fetch("jokes", "question").then((res) => res.joke),
    fetch("jokes", "answer").then((res) => res.answer),
  ]).then(([joke, answer]) => ({ question: joke, answer }));
```